packages/itwin/property-grid/src/components/MultiElementPropertyGrid.tsx

```diff
@@ -63,35 +74,168 @@ export const MultiElementPropertyGrid = (props: PropertyGridProps) => {
             });
           }
         );
+
+        // if selection is not from us, clear the ancestry
+        if (undefined === args || args.source !== ProprtyGridSelectionScope) {
+          setAncestorKeys([]);
+        }
+
         setInstanceKeys(instanceKeys);
-        setMoreThanOneElement(selectionSet.instanceKeysCount > 1);
         setSelectedInstanceKey(undefined);
       }
     };
 
+    const onFrontstageReady = () => {
+      onSelectionChange();
+    };
+
     // ensure this selection handling runs if component mounts after the selection event fires:
     onSelectionChange();
 
     const removePresentationListener = Presentation.selection.selectionChange.addListener(onSelectionChange);
     // if the frontstage changes and a selection set is already active we need to resync this widget's state with that selection
-    const removeFrontstageReadyListener = FrontstageManager.onFrontstageReadyEvent.addListener(onSelectionChange);
+    const removeFrontstageReadyListener = FrontstageManager.onFrontstageReadyEvent.addListener(onFrontstageReady);
     return () => {
       removePresentationListener();
       removeFrontstageReadyListener();
     };
   }, [iModelConnection]);
 
+  useEffect(() => {
+    // update the ancestry when the instanceKeys change
+    if (instanceKeys.length === 1) {
+      const newAncestors = [...ancestorKeys, ...instanceKeys];
+      setAncestorKeys(newAncestors);
+    }
+  }, [instanceKeys]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  useEffect(() => {
+    const getSingleElementId = (keys?: KeySet): string | undefined => {
+      let elementIds = new Set<string>();
+      if (keys?.instanceKeysCount === 1) {
+        keys.instanceKeys.forEach((ids: Set<string>) => {
+          elementIds = new Set([...elementIds, ...ids]);
+        });
+        if (elementIds.size === 1) {
+          return [...elementIds][0];
+        }
+      }
+      return undefined;
+    };
+
+    // determine if the current instance key has a parent
+    if (PropertyGridManager.flags.enableAncestorNavigation) {
+      if (iModelConnection && instanceKeys?.length === 1) {
+        const elementId = instanceKeys[0].id;
+        Presentation.selection.scopes.computeSelection(
+          iModelConnection,
+          elementId,
+          { id: "element", ancestorLevel: 1 })
+          .then((parentKeys) => {
+            setHasParent(getSingleElementId(parentKeys) !== elementId);
+          })
+          .catch(() => { });
```

Comment:
it'd be good to use a Logger.logException here in this catch

Reply:
Added Logger, not sure how I missed that.